py_scripts/parser.py

```diff
@@ -0,0 +1,203 @@
+        classification_id = classification_id.attrib.get('id','Unknown')\
+            if classification_id is not None else 'Unknown'
+        
+            
+        for clls in self.root.findall('.//ClassificationNode'): # find all the disorders, nested in the root 
+
+                        
+            disorder_id = clls.find('.//Disorder').attrib.get('id', 'unknown')\
+                if clls.find('.//Disorder') is not None else 'Unknown' # find the disorder id, nested in the disorder
+                
+            disorder_type = clls.find('.//DisorderType/Name').text\
+                if clls.find('.//DisorderType/Name') is not None else 'Unknown' # find the disorder type, nested in the disorder
+            
+            orpha_code = clls.find('.//OrphaCode').text\
+                if clls.find('.//OrphaCode') is not None else 'Unknown' # find the orpha code, nested in the disorder
+            
+            name = clls.find('.//Name').text\
+                if clls.find('.//Name') is not None else 'Unknown' # find the name, nested in the disorder
+                
+            expert_link = clls.find('.//ExpertLink').text\
+                if clls.find('.//ExpertLink') is not None else 'Unknown' # find the expert link, nested in the disorder
+            
+            disease = Disease( classification_id, disorder_type, disorder_id,\
+                            orpha_code, name, expert_link, self.meta_data)    # create a disease object
+            
+            diseases.append(disease)
+        
+        return diseases
+
+
+def get_xml_files(directory):
+    
+    return [os.path.join(root, file)\
+        for root, _, files in os.walk(directory)\
+            for file in files if file.endswith('.xml')]
+
+
+
+def process_progress_bar(xml_files): # might be useful for later
+    doty = "."
+    returny = 0
+    for xml_file in tqdm(xml_files, desc=f"Processing XML Files{doty}]"):
+        while returny <= 4 :
+            doty += "."
+            returny += 1
+            if returny == 4:
+                doty = "."
+                returny = 0
+                break
+        yield xml_file
+        
+    return xml_file
+
+
+def process_xml_files(xml_files, directory):
+    
+    for xml_file in tqdm(xml_files, desc="Processing XML Files"):
+        parser = OrphanetXMLParser(xml_file)
+        diseases = parser.parse_diseases()
+        diseases_df = pd.DataFrame([disease.to_dict() for disease in diseases])
+        
+        base_name = os.path.basename(xml_file)
+        csv_file_name = os.path.splitext(base_name)[0] + '.csv'
+        csv_file_path = os.path.join(directory, csv_file_name)
+        
+        diseases_df.to_csv(csv_file_path, index=False)
+        print(f'File saved to {csv_file_path}')
+
+
+def main():
+    os.makedirs(directory, exist_ok=True)
+    xml_files = get_xml_files(raw_data_path)
+    process_xml_files(xml_files, directory)
```

Comment:
look good but use python types line  def func(x: str) -> str:
etc

Reply:
Please check out the recent pr I am initiating, and return any insight if the structure is still needed.